test_config.py

```diff
@@ -12,8 +12,13 @@ def check_test_solver_install(solver_class):
     detecting the situation.
     """
     cuda_version = get_cuda_version()
+    is_platform_macOS = sys.platform == "darwin"
+
     if solver_class.name.lower() == "cuml":
-        if sys.platform == "darwin":
+        if is_platform_macOS:
             pytest.xfail("Cuml is not supported on MacOS.")
         if cuda_version is None:
             pytest.xfail("Cuml needs a working GPU hardware.")
+
+    if is_platform_macOS and "snapml" in solver_class.name.lower():
+        pytest.xfail("running snapml on MacOS takes a lot of time.")
```

Comment:
can you skip only the `test_solver` test?
(add a function `check_test_solver(solver_class)` in this file I believe)